### ISSUE TITLE
(Blazor WASM) returnUrl for the login action of RemoteAuthenticatorView is not honored when LoginMode=redirect

### ISSUE THREAD

**Issue by umlaufa1** [dotnet/aspnetcore#28769]
This problem applies to RemoteAuthenticatorView and  MsalProviderOptions for Blazor WASM

### Describe the bug
The returnUrl for the login action of RemoteAuthenticatorView is not honored when LoginMode property of MsalProviderOptions is set to redirect. This problem was discovered by making two small modifications to template source code obtained from this tutorial: 
https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory-b2c?view=aspnetcore-5.0

### To Reproduce
1. Download the template project and configure for a b2c account. 

`dotnet new blazorwasm -au IndividualB2C --aad-b2c-instance "{AAD B2C INSTANCE}" --api-client-id "{SERVER API APP CLIENT ID}" --app-id-uri "{SERVER API APP ID URI}" --client-id "{CLIENT APP CLIENT ID}" --default-scope "{DEFAULT SCOPE}" --domain "{TENANT DOMAIN}" -ho -o {APP NAME} -ssp "{SIGN UP OR SIGN IN POLICY}"`

2. Add the following option to AddMSALAuthentication in program.cs

```
            builder.Services.AddMsalAuthentication(options =>
            {
                ...
                options.ProviderOptions.LoginMode = "redirect";
            });
```
3.  Add the following attribute to Counter.razor 

`@attribute [Microsoft.AspNetCore.Authorization.Authorize]`

NOTE: this attribute will cause RedirectToLogin.razor to invoke Navigation.NavigateTo($"authentication/login?returnUrl={Uri.EscapeDataString(Navigation.Uri)}") when you navigate to https://localhost:port/counter

4.  In the URL field of your web browser, goto https://localhost:port/counter 

You will be redirected to the B2C login flow. After logging in, you will be redirected back to http://localhost:port instead of https://localhost:port/counter. 

NOTE: if you remove options.ProviderOptions.LoginMode = "redirect" and use popups for the B2C user flow, then you will be properly redirected back to https://localhost:port/counter after logging in.

**Comment by Akinzekeel** [comment]
I'm having the same issue but with AAD (not B2C).

I have observed the same behavior even when leaving LoginMode untouched (which defaults to popup if possible, then redirect if the popup is blocked by the browser).

I believe this is because the returnUrl is not passed on to the login provider (which afaik is the correct behavior as far as login providers are concerned). However there should be some client side code that can store the returnUrl & restore it when the user returns to authentication/login-callback.

**Comment by captainsafia** [comment]
Thanks for opening this issue. Are you able to reproduce this with .NET 5.0.2? We received a report of this bug earlier and shipped a patch in 5.0.2.

**Comment by umlaufa1** [comment]
I can confirm the bug exists with 5.0.1. I do not have access to .NET 5.0.2 (or I am not sophisticated enough to access 5.0.2 through channels other NuGet :-) ).

**Comment by captainsafia** [comment]
🤦🏽‍♀️  Just realized that 5.0.2 will be released this month.

If you're up for it, you can grab the nightly installer at https://github.com/dotnet/installer#installers-and-binaries.

Otherwise, I'd recommend seeing if this repros once 5.0.2 is out. If you still have problems, feel free to open another issue.
